rusoto/credential/src/instance_metadata.rs

```diff
@@ -48,13 +49,19 @@ impl InstanceMetadataProvider {
         InstanceMetadataProvider {
             client: HttpClient::new(),
             timeout: Duration::from_secs(30),
+            port_for_testing: None,
         }
     }
 
     /// Set the timeout on the provider to the specified duration.
     pub fn set_timeout(&mut self, timeout: Duration) {
         self.timeout = timeout;
     }
+
+    /// Set source to localhost on specified port for integration testing.
+    pub fn set_to_localhost(&mut self, port: &str) {
```

Comment:
I think ideally this wouldn't be `pub`, if we're worried about complete overriding of the instance metadata service endpoint? Could we move the `it_fetches_basic_role` test into this module to avoid that?

Reply:
That's an option! I was leaning towards an integration test because it uses the crate like external projects do, such as `rusoto_core`. 

Perhaps we can see how this could look running as a Rust unit test that's actually an integration test. 😄